### ISSUE TITLE
Would it be possible to use it in the field without internet?

### ISSUE THREAD

**Issue by jaerabr** [Stefal/rtkbase#204]
Hello, I would like to congratulate you for your work, I am using it in my rtk kit with f9p bt 300 antenna and cuav radios, I would like to know if I could use it without internet access as an rtk base in the field for topography, I did a test here it does not record raw files without clock update, I tried to install an RTC DS3231 module but I can't configure
Thanks sorry for the google translation

Thanks for replying, I'm using the Raspberry Pi Zero W Rev 1.1 but I turned it to work in access point mode.

login as: basegnss
basegnss@192.168.4.1's password:
Linux basegnss 5.10.60+ #1449 Wed Aug 25 14:59:20 BST 2021 armv6l

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
Last login: Sat Feb 26 21:51:03 2022 from 192.168.4.14
basegnss@basegnss:~ $ sudo systemctl status gpsd
sudo: unable to resolve host basegnss: Temporary failure in name resolution
● gpsd.service - GPS (Global Positioning System) Daemon
   Loaded: loaded (/etc/systemd/system/gpsd.service; enabled; vendor preset: enabled)
   Active: failed (Result: signal) since Sat 2022-02-26 21:48:40 CET; 5min ago
  Process: 494 ExecStart=/usr/sbin/gpsd $GPSD_OPTIONS $OPTIONS $DEVICES (code=killed, signal=ILL)

Feb 26 21:48:40 basegnss systemd[1]: Starting GPS (Global Positioning System) Daemon...
Feb 26 21:48:40 basegnss systemd[1]: gpsd.service: Control process exited, code=killed, status=4/ILL
Feb 26 21:48:40 basegnss systemd[1]: gpsd.service: Failed with result 'signal'.
Feb 26 21:48:40 basegnss systemd[1]: Failed to start GPS (Global Positioning System) Daemon.
basegnss@basegnss:~ $ chronyc sources -v
210 Number of sources = 1

  .-- Source mode  '^' = server, '=' = peer, '#' = local clock.
 / .- Source state '*' = current synced, '+' = combined , '-' = not combined,
| /   '?' = unreachable, 'x' = time may be in error, '~' = time too variable.
||                                                 .- xxxx [ yyyy ] +/- zzzz
||      Reachability register (octal) -.           |  xxxx = adjusted offset,
||      Log2(Polling interval) --.      |          |  yyyy = measured offset,
||                                \     |          |  zzzz = estimated error.
||                                 |    |           \
MS Name/IP address         Stratum Poll Reach LastRx Last sample
===============================================================================
#? GNSS                          0   4     0     -     +0ns[   +0ns] +/-    0ns
basegnss@basegnss:~ $

**Comment by jaerabr** [comment]
Thanks for replying, I'm using the Raspberry Pi Zero W Rev 1.1 but I turned it to work in access point mode.

login as: basegnss
basegnss@192.168.4.1's password:
Linux basegnss 5.10.60+ #1449 Wed Aug 25 14:59:20 BST 2021 armv6l

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
Last login: Sat Feb 26 21:51:03 2022 from 192.168.4.14
basegnss@basegnss:~ $ sudo systemctl status gpsd
sudo: unable to resolve host basegnss: Temporary failure in name resolution
● gpsd.service - GPS (Global Positioning System) Daemon
   Loaded: loaded (/etc/systemd/system/gpsd.service; enabled; vendor preset: enabled)
   Active: failed (Result: signal) since Sat 2022-02-26 21:48:40 CET; 5min ago
  Process: 494 ExecStart=/usr/sbin/gpsd $GPSD_OPTIONS $OPTIONS $DEVICES (code=killed, signal=ILL)

Feb 26 21:48:40 basegnss systemd[1]: Starting GPS (Global Positioning System) Daemon...
Feb 26 21:48:40 basegnss systemd[1]: gpsd.service: Control process exited, code=killed, status=4/ILL
Feb 26 21:48:40 basegnss systemd[1]: gpsd.service: Failed with result 'signal'.
Feb 26 21:48:40 basegnss systemd[1]: Failed to start GPS (Global Positioning System) Daemon.
basegnss@basegnss:~ $ chronyc sources -v
210 Number of sources = 1

  .-- Source mode  '^' = server, '=' = peer, '#' = local clock.
 / .- Source state '*' = current synced, '+' = combined , '-' = not combined,
| /   '?' = unreachable, 'x' = time may be in error, '~' = time too variable.
||                                                 .- xxxx [ yyyy ] +/- zzzz
||      Reachability register (octal) -.           |  xxxx = adjusted offset,
||      Log2(Polling interval) --.      |          |  yyyy = measured offset,
||                                \     |          |  zzzz = estimated error.
||                                 |    |           \
MS Name/IP address         Stratum Poll Reach LastRx Last sample
===============================================================================
#? GNSS                          0   4     0     -     +0ns[   +0ns] +/-    0ns
basegnss@basegnss:~ $

**Comment by Stefal** [comment]
Ok!
There is a problem with the gpsd backport for the Raspberry Pi Zero. See #175 
A workaround is to compile/install gpsd from the sources.
Another solution could be to try with raspberry pi Os lite "bullseye"

**Comment by jaerabr** [comment]
I did the test with the raspberry pi Os lite "bullseye" I leave it off the internet it doesn't save the raw file, as soon as I connect to the internet it simply starts recording raw data.

**Comment by Stefal** [comment]
Don't buy an Orange Pi Zero only for this, it won't fix the problem.

I did some tests on a Raspberry Pi Zero V1.1 today. With the bullseye release, gpsd is successfuly installed, but it doesn't start correctly. You can try on your side with the 2.3.3 update, but I think it won't work as it should.
At least, you can check that gpsd is correctly installed with `sudo gpsd -V`

If you try `sudo systemctl status gpsd` please tell me if you have the message as me:
```
Feb 28 17:37:48 raspberrypi systemd[1]: gpsd.service: Bound to unit str2str_tcp.service, but unit isn't active.
Feb 28 17:37:48 raspberrypi systemd[1]: Dependency failed for GPS (Global Positioning System) Daemon.
Feb 28 17:37:48 raspberrypi systemd[1]: gpsd.service: Job gpsd.service/start failed with result 'dependency'.
```

**Comment by Stefal** [comment]
If you have a bullseye release on your Pi Zero v1, and RTKBase v2.3.3 try this:
Edit the gpsd unit with `sudo nano /etc/systemd/system/gspd.service`
The unit should contains this:
```
[Unit]
Description=GPS (Global Positioning System) Daemon
Requires=gpsd.socket
After=str2str_tcp.service

[Service]
Type=forking
EnvironmentFile=-/etc/default/gpsd
ExecStartPre=systemctl is-active str2str_tcp.service
ExecStart=/usr/sbin/gpsd $GPSD_OPTIONS $OPTIONS $DEVICES
Restart=always
RestartSec=30

[Install]
WantedBy=multi-user.target
Also=gpsd.socket
```
Save (Ctrl + x) Then execute `sudo systemctl daemon-reload` and reboot.
After the reboot, start the RTKBase main service and File service, and wait a few minutes.

As soon as `chronyc sources` shows you a * in front of the GNSS source like this:
```
#* GNSS                          0   4   377    19    -62us[-2445us] +/-  200ms
```
Then the File service should start to record the Gnss data.

**Comment by jaerabr** [comment]
> 
Hello, I followed the steps here and now it works perfectly without internet, obs: there is a typo in "gspd.service"

- basegnss@basegnss:~ $ chronyc sources
MS Name/IP address         Stratum Poll Reach LastRx Last sample
===============================================================================
#* GNSS                          0   4   377     8   +509us[ +722us] +/-  200ms
^? 192.0.2.1                     0   8     0     -     +0ns[   +0ns] +/-    0ns
basegnss@basegnss:~ $

thanks for the quick reply friend

**Comment by jaerabr** [comment]
Hello, I followed the steps here and now it works perfectly without internet, obs: there is a typo in "gspd.service"

- basegnss@basegnss:~ $ chronyc sources
MS Name/IP address Stratum Poll Reach LastRx Last sample
===============================================================================
#* GNSS 0 4 377 8 +509us[ +722us] +/- 200ms
^? 192.0.2.1 0 8 0 - +0ns[ +0ns] +/- 0ns
basegnss@basegnss:~ $

sometimes it stops recording, you have to restart everything to come back I will continue monitoring here.
thanks for the quick reply friend

**Comment by Stefal** [comment]
> Hello, I followed the steps here and now it works perfectly without internet
That's a good news !!
> obs: there is a typo in "gspd.service"
Thank! Fixed!
> sometimes it stops recording, you have to restart everything to come back I will continue monitoring here. thanks for the quick reply friend
The service which store the data inside your Pi Zero check if the internal datetime is synced before starting to record, but only before, not during the logging. If it stopped, the reason is probably elsewhere.

**Comment by Stefal** [comment]
@jaerabr Any update ?

**Comment by jaerabr** [comment]
> @jaerabr Alguma atualização?

good afternoon for me it works well thank you
problem solved

**Comment by Stefal** [comment]
Thanks ! This fix will be included in the next release.
